cse-lib/network-processing/src/main/java/com/farao_community/farao/cse/network_processing/busbar_change/NetworkModifier.java

```diff
@@ -0,0 +1,316 @@
+        connectablesToRemove.add(transformer);
+        return newTransformer;
+    }
+
+    private static void copyTapChanger(TwoWindingsTransformer transformerFrom, TwoWindingsTransformer transformerTo) {
+        PhaseTapChanger pst = transformerFrom.getPhaseTapChanger();
+        if (pst == null) {
+            return;
+        }
+        PhaseTapChangerAdder ptca = transformerTo.newPhaseTapChanger()
+            .setLowTapPosition(pst.getLowTapPosition())
+            .setTapPosition(pst.getTapPosition())
+            .setRegulationValue(pst.getRegulationValue())
+            .setRegulationMode(pst.getRegulationMode());
+        pst.getAllSteps().values().forEach(step ->
+            ptca.beginStep()
+                .setRho(step.getRho())
+                .setAlpha(step.getAlpha())
+                .setR(step.getR())
+                .setX(step.getX())
+                .setG(step.getG())
+                .setB(step.getB())
+                .endStep()
+        );
+        ptca.add();
+    }
+
+    private static void copyProperties(Identifiable<?> identifiableFrom, Identifiable<?> identifiableTo) {
+        identifiableFrom.getPropertyNames().forEach(property ->
+            identifiableTo.setProperty(property, identifiableFrom.getProperty(property))
+        );
+    }
+
+    private LineAdder initializeLineAdderToMove(Line line, String newId) {
+        return network.newLine()
+            .setId(newId)
+            .setR(line.getR())
+            .setX(line.getX())
+            .setG1(line.getG1())
+            .setB1(line.getB1())
+            .setG2(line.getG2())
+            .setB2(line.getB2())
+            .setFictitious(line.isFictitious())
+            .setName(newId);
+    }
+
+    private static BranchAdder<?> setIdenticalToSide(Branch<?> branch, Branch.Side side, BranchAdder<?> adder) {
+        TopologyKind topologyKind = branch.getTerminal(side).getVoltageLevel().getTopologyKind();
+        if (topologyKind == TopologyKind.BUS_BREAKER) {
+            if (side == Branch.Side.ONE) {
+                return adder.setVoltageLevel1(branch.getTerminal1().getVoltageLevel().getId())
+                    .setConnectableBus1(branch.getTerminal1().getBusBreakerView().getConnectableBus().getId())
+                    .setBus1(branch.getTerminal1().getBusBreakerView().getBus() != null ? branch.getTerminal1().getBusBreakerView().getBus().getId() : null);
+            } else if (side == Branch.Side.TWO) {
+                return adder.setVoltageLevel2(branch.getTerminal2().getVoltageLevel().getId())
+                    .setConnectableBus2(branch.getTerminal2().getBusBreakerView().getConnectableBus().getId())
+                    .setBus2(branch.getTerminal2().getBusBreakerView().getBus() != null ? branch.getTerminal2().getBusBreakerView().getBus().getId() : null);
+            }
+        }
+        throw new AssertionError();
```

Comment:
Throw FaraoException or just remove the if and the exception

Reply:
ok